### PR TITLE
Refactor out components shared with Workplace Search

### PR DESCRIPTION
**Pull request by cee-chen**
This PR pulls out various shared helpers (getUserName, and a new SetupGuide layout/component) that Workplace Search will be using as well. Example:

Screen Shot 2020-06-16 at 3 08 32 PM

@scottybollinger, let me know if there's any others you'd like me to add while I'm here (maybe the error messages? Although I think you mentioned those maybe don't share as much in common as we thought)

### CODE REVIEW

File: x-pack/plugins/enterprise_search/public/applications/app_search/components/setup_guide/setup_guide.tsx [hunk 1]
```diff
@@ -5,237 +5,58 @@
...
-      </EuiPageBody>
-    </EuiPage>
-  );
-};
+export const SetupGuide: React.FC = () => (
+  <SetupGuideLayout
+    productName={i18n.translate('xpack.appSearch.productName', { defaultMessage: 'App Search' })}
+    productEuiIcon="logoAppSearch"
+    standardAuthLink="https://swiftype.com/documentation/app-search/self-managed/security#standard"
+    elasticsearchNativeAuthLink="https://swiftype.com/documentation/app-search/self-managed/security#elasticsearch-native-realm"
```
Comment (cee-chen) [inline x-pack/plugins/enterprise_search/public/applications/app_search/components/setup_guide/setup_guide.tsx:22]: I'm hoping we can potentially lose these two per-product props if I get approval from @chriscressman for opening up a PR for pulling out

https://www.elastic.co/guide/en/app-search/current/security-and-users.html
and
https://www.elastic.co/guide/en/workplace-search/current/workplace-search-security.html

into its own section in the general https://www.elastic.co/guide/en/enterprise-search/current/index.html docs 🤞

**Review by scottybollinger — APPROVED** [review]
@constancecchen thanks for doing this! There's a case to be made to share the `error_state` and `no_user_state` components, as they are identical, except for the product name. For both, if we do decide to share them, we could just share the `EuiEmptyPrompt` since the breadcrumbs, telemetry and page header are different. I'm on the fence about the benefit of sharing these, but since these should be used post-MVP, I thought it would be good to get your thoughts.

**Comment by cee-chen** [comment]
Ahh! So actually, I believe the `no_user_state` component will be replaced by the "don't show AS/WS at all in the nav if the user has no access" functionality that Brandon K is asking us to implement.

That leaves only the error state left, and thinking more about it, I'll most likely end up refactoring that out to a kind of shared interceptor of some kind (either in the client side or server side) so that each page isn't in charge of checking for errors / showing the component.

With that in mind, I'll probably leave the states alone for now since they're due to be refactored later, if that sounds cool!

File: x-pack/plugins/enterprise_search/public/applications/app_search/components/setup_guide/setup_guide.tsx (hunk 1 above)
Comment (chriscressman) [reply x-pack/plugins/enterprise_search/public/applications/app_search/components/setup_guide/setup_guide.tsx:22]: @constancecchen - I'm not sure if you'd still like to make this change, but if so, I welcome that PR. I'm tracking a larger, related docs project in https://github.com/elastic/enterprise-search-pubs/issues/630, but I don't expect to act on it for a while. More broadly, I favor moving shared concerns out of product-specific books and into the Enterprise Search book.

If you proceed, here are some considerations:

- @jonas updated the Workplace Search version of the doc for 7.7.0, so it's more likely to be up to date than the App Search version.
- The existing Workplace Search URL will need to be redirected. In the PR, you can do a manual redirect (have the old doc link to the new), and then after it's deployed, I can get the website team to implement an HTTP redirect.
- The existing Workplace Search page is part of an "Administration & management" section, so you may need to modify the contents of the other docs in that section if they depend on the doc that's moving (I haven't checked).
- You'll have to find any links to the product-specific docs in the App Search and Workplace Search books and update them to point to the Enterprise Search book.

I wish moving docs wasn't so hard :cry:

**Pull request merged**